### ISSUE TITLE
Simplify error parsing

### ISSUE THREAD

**Issue by garrytrue** [haroldadmin/NetworkResponseAdapter#10]
I think _HttpException.extractFromHttpException_ can return  _NetworkResponse.ServerError<E>_ instead if _NetworkResponse_.

And one more, you have broken test in _ErrorExtractionTest_

If you allow me I can make a PR with fixes

**Comment by haroldadmin** [comment]
Thank you for pointing out the issue. I looked into the testing setup as a result and as it turns out, most of the tests are not being picked up by Gradle. 

The project right now has 42 tests, but running `./gradlew clean test` only executes 18 of them. This is why I missed the broken test in `ErrorExtractionTest`. I have found there are a few other broken tests too which need fixing.

The reason is that some test classes do not extend from a base class of `kotlintest` library. I am working on a fix for this.

Regarding the `HttpException.extractFromHttpException` point, I think you're right. An HTTP exception by definition indicates that an error response was received from the server, so classifying it as something other than a `ServerError` would be silly. Feel free to make a PR for this, I am happy to accept external contributions.
